Update release documentation for stable branches (#430)
submariner-io/submariner-website: src/content/development/release-process/_index.en.md

Sometimes you may want to create a specific project release for testing prior to creating a release candidate. In this
case, the suggested naming convention is to append `-preN`.

### Stable branches and versions

Stable branches will be maintained for `vx.0` versions and any derived versions. For example, versions v0.8.0, v0.8.1 and

> any other minor release will all be maintained and released on the `release-0.8` branch.

Reviewer: ```suggestion
any other minor releases will all be maintained and released on the `release-0.8` branch.
```

> Stable branches will be maintained for `vx.0` versions and any derived versions. For example, versions v0.8.0, v0.8.1 and
> any other minor releases will all be maintained and released on the `release-0.8` branch.

Reviewer: ```suggestion
Stable branches will be maintained for major and minor versions. For example, versions v0.8.0, v0.8.1 and
any other 0.8.x releases will all be maintained and released on the `release-0.8` branch.
```

> The automated release process, which is described henceforth, can also create these stable branches when necessary.

Reviewer: ```suggestion
The automated release process described here can also create these stable branches when necessary.
```

green check mark on success or a red X on failure. A failure likely means the artifacts were not published to Quay, in
which case select the failed check, inspect the logs, correct the issue and re-run the job.

### Step 0: Create a stable branch (if necessary)

> In case you're creating a new stable release, the release automation process can create the stable branches. To do so,

Reviewer: ```suggestion
If you're creating a new stable release, the release automation process can create the stable branches. To do so,
```

navigate to the [releases](https://github.com/submariner-io/releases) repository and fork it.

1) Create a new branch for the intended release.

> 2) Create a new file in the `releases` directory (you can copy the `example.yaml file`). For our example, we'll name it `v0.8.0.yaml`.

Reviewer: ```suggestion
2) Create a new file in the `releases` directory (you can copy the `example.yaml` file). For our example, we'll name it `v0.8.0.yaml`.
```
